nixos/modules/services/web-apps/tachidesk-server.nix

```
@@ -0,0 +1,254 @@
+              };
+
+              localSourcePath = mkOption {
+                type = types.path;
+                default = cfg.dataDir;
+                defaultText = lib.literalExpression "tachidesk-server.dataDir";
+                example = "/var/data/local_mangas";
+                description = mdDoc ''
+                  Path to the local source folder.
+                '';
+              };
+
+              systemTrayEnabled = mkOption {
+                type = types.bool;
+                default = false;
+                description = mdDoc ''
+                  Whether to enable a system tray icon, if possible.
+                '';
+              };
+            };
+          };
+        };
+        description = mdDoc ''
+          Configuration to write to {file}`server.conf`.
+          See <https://github.com/Suwayomi/Tachidesk-Server/wiki/Configuring-Tachidesk-Server> for more information.
+        '';
+        default = { };
+        example = {
+          server.socksProxyEnabled = true;
+          server.socksProxyHost = "yourproxyhost.com";
+          server.socksProxyPort = "8080";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [{
+      assertion = with cfg.settings.server; basicAuthEnabled -> (basicAuthUsername != null && basicAuthPasswordFile != null);
+      message = ''
+        [tachidesk-server]: the username and the password file cannot be null when the basic auth is enable
+      '';
+    }];
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.config.port ];
+
+    users.groups = mkIf (cfg.group == "tachidesk") {
+      tachidesk = { };
+    };
+
+    users.users = mkIf (cfg.user == "tachidesk") {
+      tachidesk = {
+        group = cfg.group;
+        # Need to set the user home because the package writes to ~/.local/Tachidesk
+        home = cfg.dataDir;
+        description = "Tachidesk Daemon user";
+        isSystemUser = true;
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}/.local/share/Tachidesk' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.tachidesk-server =
+      let
+        flattenConfig = prefix: config:
+          lib.foldl'
+            lib.mergeAttrs
+            { }
+            (lib.attrValues
+              (lib.mapAttrs
+                (k: v:
+                  if !(lib.isAttrs v)
+                  then { "${prefix}${k}" = v; }
+                  else flattenConfig "${prefix}${k}." v
+                )
+                config
+              )
+            );
+
+        #  HOCON is a JSON superset that tachidesk-server use for configuration
+        toHOCON = attr:
+          let
+            attrType = builtins.typeOf attr;
+          in
+          if builtins.elem attrType [ "string" "path" "int" "float" ]
+          then ''"${toString attr}"''
+          else if attrType == "bool"
+          then lib.boolToString attr
+          else # attrs, list, lambda, null
```

Personally, I think the if else ladders are a bit easier to read when you indent the outcomes. But there's no hard requirements to style atm.

```suggestion
          if builtins.elem attrType [ "string" "path" "int" "float" ]
            then ''"${toString attr}"''
          else if attrType == "bool"
            then lib.boolToString attr
          else # attrs, list, lambda, null
```